docs/book/src/clusterctl/overview.md

```diff
@@ -5,14 +5,16 @@ The `clusterctl` CLI tool handles the lifecycle of a Cluster API [management clu
 The `clusterctl` command line interface is specifically designed for providing a simple "day 1 experience" and a
 quick start with Cluster API; it automates fetching the YAML files defining [provider components] and installing them.
 
-Additionally it encodes a set of best practices in managing providers, that helps the user in avoiding 
+Additionally it encodes a set of best practices in managing providers, that helps the user in avoiding
 mis-configurations or in managing day 2 operations such as upgrades.
 
 * use [`clusterctl init`](commands/init.md) to install Cluster API providers
 * use [`clusterctl upgrade`](commands/upgrade.md) to upgrade Cluster API providers
 * use [`clusterctl delete`](commands/delete.md) to delete Cluster API providers
 
 * use [`clusterctl config cluster`](commands/config-cluster.md) to spec out workload clusters
+* use [`clusterctl generate yaml`](commands/generate-yaml.md) to process yaml
```

Comment:
Is the idea to consolidate generate and config later on? I imagine there is some overlap between the two (eg. --list-variables). It might be confusing to a user when to use one vs. the other

Reply:
We have this issue documenting some of the refactors we'd like to have for the config sub-command. https://github.com/kubernetes-sigs/cluster-api/issues/3360

Reply:
Got it. Also in the issue for user story you wrote "As a developer I would like to have a sub-command in clusterctl that could process yaml using the built in yaml processor so that we can leverage clusterctl's internal yaml processor within scripts." I agree that the main audience for this command would be a developer, do you also foresee a user story for cluster api end users? If not, would it be confusing to users to include this one in the user documentation? Maybe we can add something like "for development purposes" in the description, wdyt?

Reply:
These could also be used in CI scripts and other custom pipelines

Reply:
@CecileRobertMichon Yes. That makes sense actually. I guess for this specific sub-command the main audience would be for developers or maintainers of cluster-api providers (from the use cases we've seen so far). 

Unless there are users who have customized templates and want to use clusterctl's default variable processor as well.
I can try and add some more clarity towards this in the book documentation.